crates/context/config/src/client/starknet.rs

```diff
@@ -158,6 +167,30 @@ impl Transport for StarknetTransport<'_> {
     }
 }
 
+fn compute_transaction_hash(
+    sender_address: Felt,
+    contract_address: Felt,
+    entry_point_selector: Felt,
+    calldata: &[Felt],
+) -> Felt {
+    let elements: Vec<&Felt> = vec![&sender_address, &contract_address, &entry_point_selector]
+        .into_iter()
+        .chain(calldata.iter())
+        .collect();
+
+    poseidon_hash_many(elements)
```

Comment:
you shouldn't need to allocate - no `vec!` or `collect` necessary

```suggestion
    let elements = [sender_address, contract_address, entry_point_selector]
        .iter()
        .chain(calldata.iter());

    poseidon_hash_many(elements)
```

Reply:
Updated